### PR TITLE
fix: phone number compulsory field remove

### PR DESCRIPTION
**Pull request by Parth910**
Fixes ##3289

#### Short description of what this resolves:
In the profile section, a phone number was a compulsory  field, so I made it an optional field

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Parth910** [comment]
@kushthedude  yes that's true, but here it was the compulsory field made by validation so no other way to make it optional.

**Comment by kushthedude** [comment]
> @kushthedude yes that's true, but here it was the compulsory field made by validation so no other way to make it optional.

Are you sure ?

File: app/components/forms/user-profile-form.js [hunk 1]
```diff
@@ -41,20 +40,6 @@ export default Component.extend(FormMixin, {
...
-        phone: {
-          identifier : 'phone',
-          rules      : [
-            {
-              type   : 'empty',
-              prompt : this.l10n.t('Please enter a phone number.')
-            },
-            {
-              type   : 'regExp',
-              value  : validPhoneNumber,
```
Comment (shreyanshdwivedi) [inline app/components/forms/user-profile-form.js]: why remove this?

File: app/components/forms/user-profile-form.js [hunk 2]
```diff
@@ -41,20 +40,6 @@ export default Component.extend(FormMixin, {
               prompt : this.l10n.t('Please enter a valid email ID')
             }
           ]
-        },
```
Comment (kushthedude) [inline app/components/forms/user-profile-form.js]: @Parth910 Just add
```js
optional  : true,
```
Rest is done

Comment (Parth910) [reply app/components/forms/user-profile-form.js]: Thank you for help, I made some unwanted changes

File: app/components/forms/user-profile-form.js (hunk 1 above)
Comment (Parth910) [reply app/components/forms/user-profile-form.js]: I think that after removing an "empty" type, a validator can't allow empty input in the phone field. so I just remove it and add inputtype=number.

**Comment by Parth910** [comment]
> Dont change the other linea
 ok

File: app/components/forms/user-profile-form.js [hunk 3]
```diff
@@ -44,15 +44,16 @@ export default Component.extend(FormMixin, {
         },
         phone: {
           identifier : 'phone',
+          optional   : true,
           rules      : [
             {
               type   : 'empty',
```
Comment (CosmicCoder96) [inline app/components/forms/user-profile-form.js]: remove the empty rule completely, optional and empty both don't make sense.

Comment (Parth910) [reply app/components/forms/user-profile-form.js]: ok

File: app/components/forms/user-profile-form.js [hunk 4]
```diff
@@ -44,15 +44,12 @@ export default Component.extend(FormMixin, {
         },
         phone: {
           identifier : 'phone',
+          optional   : true,
```
Comment (shreyanshdwivedi) [inline app/components/forms/user-profile-form.js]: Do we need optional here?

**Comment by mrsaicharan1** [comment]
@Parth910 Just a pointer. Make sure that you link the issues properly. This helps in closing the issue automatically.

**Pull request merged**